foursight_core/react/api/aws_stacks.py

```diff
@@ -0,0 +1,133 @@
+    including various metadata, outputs, parameters, and resources.
+    Output and parameter values are obfuscated if their names represents a sensitive value.
+    """
+    result = {}
+    stack = _aws_get_stack_object(stack_name)
+    if stack:
+        result = {**_create_aws_stack_info(stack)}
+        outputs = aws_get_stack_outputs(stack)
+        parameters = aws_get_stack_parameters(stack)
+        resources = aws_get_stack_resources(stack)
+        result["outputs"] = outputs
+        result["parameters"] = parameters
+        result["resources"] = resources
+        pass
+    return result
+
+
+@memoize
+def aws_get_stack_outputs(stack_name: str) -> dict:
+    """
+    Returns the name/value outputs for the given AWS CloudFormation stack name.
+    Output values are obfuscated if the output name represents a sensitive value.
+    """
+    result = {}
+    stack = _aws_get_stack_object(stack_name)
+    if stack and stack.outputs:
+        for stack_output in sorted(stack.outputs, key=lambda key: key["OutputKey"]):
+            result[stack_output.get("OutputKey")] = stack_output.get("OutputValue")
+    return obfuscate_dict(result, obfuscated="********")
+
+
+@memoize
+def aws_get_stack_parameters(stack_name: str) -> dict:
+    """
+    Returns a name/value dictionary of the parameters for the given AWS CloudFormation stack name.
+    Parameter values are obfuscated if the parameter name represents a sensitive value.
+    """
+    result = {}
+    stack = _aws_get_stack_object(stack_name)
+    if stack and stack.parameters:
+        for stack_parameter in sorted(stack.parameters, key=lambda key: key["ParameterKey"]):
+            result[stack_parameter.get("ParameterKey")] = stack_parameter.get("ParameterValue")
+    return obfuscate_dict(result, obfuscated="********")
+
+
+@memoize
+def aws_get_stack_resources(stack_name: str) -> dict:
+    """
+    Returns a name/value dictionary of the resources for the given AWS CloudFormation stack name.
+    """
+    result = {}
+    stack = _aws_get_stack_object(stack_name)
+    if stack:
+        for stack_resource in sorted(list(stack.resource_summaries.all()), key=lambda key: key.logical_resource_id):
+            result[stack_resource.logical_resource_id] = stack_resource.resource_type
+    return result
+
+
+def _aws_get_stack_object(stack_name_or_object: Union[str, object]) -> Optional[object]:
+    if "cloudformation.Stack" in str(type(stack_name_or_object)):  # TODO: right way to type check.
+        return stack_name_or_object
+    c4 = boto3.resource("cloudformation")
+    stack = list(c4.stacks.filter(StackName=stack_name_or_object))
+    return stack[0] if len(stack) == 1 else None
+
+
+@memoize
+def aws_get_stack_template(stack_name: str) -> str:
+    c4 = boto3.client("cloudformation")
+    stack_template = c4.get_template(StackName=stack_name)
+    stack_template_body = stack_template["TemplateBody"]
+    if isinstance(stack_template_body, OrderedDict):
```

Comment:
I actually changed this to return a dictionary, so that I could obfuscate (via obfuscation_utils); had previously had problems loading the YAML string into a dictionary, but I resolved that. But get your point, checking for dict or OrderedDict and otherwise assume string for YAML load into dict.